test/system/docker_live_restore_test.go

```diff
@@ -0,0 +1,172 @@
+// debug, as it can be copy-pasted to a file and applied manually.
+const dockerLiveRestoreOnNodesDSManifest = `apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  generateName: test-live-restore-
+spec:
+  selector:
+    matchLabels:
+      name: test-live-restore
+  template:
+    metadata:
+      labels:
+        name: test-live-restore
+    spec:
+      tolerations:
+      - key: ""
+        operator: Exists
+      terminationGracePeriodSeconds: 1
+      initContainers:
+      - name: test-live-restore
+        image: quay.io/kinvolk/docker:stable
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - 'docker info -f {{.LiveRestoreEnabled}} | grep true'
+        volumeMounts:
+        - name: docker-socket
+          mountPath: /var/run/docker.sock
+
+      containers:
+      - name: wait
+        image: quay.io/kinvolk/docker:stable
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - sleep infinity
+      volumes:
+      - name: docker-socket
+        hostPath:
+          path: /var/run/docker.sock
+`
+
+const dockerLogOptsOnNodesDSManifest = `apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  generateName: test-log-opts-
+spec:
+  selector:
+    matchLabels:
+      name: test-log-opts
+  template:
+    metadata:
+      labels:
+        name: test-log-opts
+    spec:
+      tolerations:
+      - key: ""
+        operator: Exists
+      terminationGracePeriodSeconds: 1
+      containers:
+      - name: wait
+        image: quay.io/kinvolk/jq:1.6
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - sleep infinity
+
+      initContainers:
+      - name: test-log-opts-1
+        image: quay.io/kinvolk/jq:1.6
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - jq '."log-opts"."max-size"' /etc/docker/daemon.json | grep 100m
+        volumeMounts:
+        - name: docker-config
+          mountPath: /etc/docker/daemon.json
+      - name: test-log-opts-2
+        image: quay.io/kinvolk/jq:1.6
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - jq '."log-opts"."max-file"' /etc/docker/daemon.json | grep 3
+        volumeMounts:
+        - name: docker-config
+          mountPath: /etc/docker/daemon.json
+      volumes:
+      - name: docker-config
+        hostPath:
+          path: /etc/docker/daemon.json
+`
+
+func TestDockerNodeConfig(t *testing.T) {
+	tcs := []struct {
+		name   string
+		config string
+	}{
+		{"Docker Live Restore", dockerLiveRestoreOnNodesDSManifest},
+		{"Docker Log Options", dockerLogOptsOnNodesDSManifest},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			namespace := "kube-system"
+			client := testutil.CreateKubeClient(t)
```

Comment:
Client can be shared between tests.

Reply:
Maybe change it to `dsClient := client.AppsV1().DaemonSets("kube-system")`, so we can avoid variable for `kube-system` and shorten lines, which improves readability.

Reply:
`namespace` var is used in one more place. I will keep that var as it is.